### PR TITLE
Implement dtype attribute access and comparison

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title. This PR implements access to `dtype` attribute for columns and arrays as well as an optimization which prunes dtype comparison branches at compile time.

### CODE REVIEW

File: rbc/tests/test_omnisci_column_basic.py [hunk 1]
```diff
@@ -550,6 +550,10 @@ def test_column_aggregate(omnisci, prop, oper):
     omnisci.reset()
     omnisci.register()
 
+
+    if prop == 'groupby' and oper == 'corr':
+        pytest.skip('omniscidb server crashes')
```
Comment (pearu) [inline rbc/tests/test_omnisci_column_basic.py:554]: This is likely omniscidb version dependent. Could you create an issue about the crash and related details what are available?

**Review by pearu — CHANGES_REQUESTED** [review]
Nice work @guilhermeleobas !

Could you also add a test where the `.dtype` is used as an input dtype argument. For instance:
```python
@omnisci('T[](T[])', T=['int32', 'int64'])
def zeros_like(x):
    z = zeros(x.size, x.dtype)
    return z
```

File: rbc/tests/test_omnisci_column_basic.py (hunk 1 above)
Comment (guilhermeleobas) [reply rbc/tests/test_omnisci_column_basic.py:554]: I've seen this failure in other builds. It seems to be a random thing.

**Review by pearu — APPROVED** [review]
LGTM when CI will be green. Thanks, @guilhermeleobas !

**Pull request merged**